packages/calimero-sdk/src/wallets/StarknetLogin/StarknetRootKey.tsx

```diff
@@ -0,0 +1,394 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { randomBytes } from 'crypto';
+import { getOrCreateKeypair } from '../../auth/ed25519';
+
+import apiClient from '../../api';
+import {
+  EthSignatureMessageMetadata,
+  NodeChallenge,
+  Payload,
+  RootKeyRequest,
+  SignatureMessage,
+  SignatureMessageMetadata,
+  WalletMetadata,
+  WalletSignatureData,
+} from '../../api/nodeApi';
+import { ResponseData } from '../../types/api-response';
+import { setStorageNodeAuthorized } from '../../storage/storage';
+import { Loading } from '../loading/Loading';
+import { getWalletType } from '../eth/type';
+import { getStarknet, StarknetWindowObject } from 'get-starknet-core';
+
+interface StarknetRootKeyProps {
+  contextId?: string;
+  rpcBaseUrl: string;
+  successRedirect: () => void;
+  navigateBack?: () => void | undefined;
+}
+
+export function StarknetRootKey({
```

Comment:
`props : StarknetRootKeyProps` so all props are automatically there and it is easier to recognize what is input

Reply:
We decided to leave this like this since we are using Callbacks because of which we would need to destructure it anyway before using them as dependencies.